src/basic_server_config.cpp

```diff
@@ -44,68 +36,50 @@ bool BasicServerConfig::InitPortNumber(NginxConfig* config) {
   return true;
 }
 
-bool BasicServerConfig::InitEchoPath(NginxConfig* config) {
-  echo_path_ = GetStatementValue(config, ECHO_KEY);
-  return true;
-}
-
-bool BasicServerConfig::InitRootUrlPaths(NginxConfig* config) {
+bool BasicServerConfig::InitRequestHandlers(NginxConfig* config) {
   std::vector<std::shared_ptr<NginxConfigStatement>> location_matches = 
     FilterStatements(config, LOCATION_OBJ);
   
-  if (location_matches.size() == 0) {
-    return false;
-  }
-  
   for (const auto& statement : location_matches) {
-    if (statement->tokens_.size() >= 2) {
+    if (statement->tokens_.size() >= 3) {
       NginxConfig* location_map_block = statement->child_block_.get();
 
-      std::string user_url = statement->tokens_[1];
-      std::string host_url = GetStatementValue(location_map_block, ROOT_KEY);
+      std::string uri = statement->tokens_[1];
+      std::string handler_id = statement->tokens_[2];
+      std::string root_dir = GetStatementValue(location_map_block, ROOT_KEY);
```

Comment:
Yeah that's true. Didn't account for potentially more things being in the child block at a later time. Fixed.